steps/progress-bar/ProgressBar.step.js

```
@@ -0,0 +1,87 @@
+/* eslint-disable operator-linebreak */
+const { Interstitial } = require('@hmcts/one-per-page/steps');
+const logger = require('@hmcts/nodejs-logging').Logger.getLogger(__filename);
+const config = require('config');
+const idam = require('services/idam');
+const { CaseStates } = require('const');
+
+const progressStates = {
+  progressedNoAos: 'progressedNoAos',
+  progressedUndefended: 'progressedUndefended',
+  awaitingAnswer: 'awaitingAnswer',
+  defendedDivorce: 'defendedDivorce'
+};
+
+const values = {
+  yes: 'yes',
+  no: 'no'
+};
+
+class ProgressBar extends Interstitial {
+  static get path() {
+    return config.paths.progressBar;
+  }
+
+  get session() {
+    return this.req.session;
+  }
+
+  get progressStates() {
+    return progressStates;
+  }
+
+  getProgressBarContent() {
+    const caseState = this.session.caseState;
+
+    if (this.progressedNoAos(caseState)) {
+      return this.progressStates.progressedNoAos;
+    } else if (this.progressedUndefended(caseState)) {
+      return this.progressStates.progressedUndefended;
+    } else if (this.awaitingAnswer(caseState)) {
+      return this.progressStates.awaitingAnswer;
+    } else if (this.defendedDivorce(caseState)) {
+      return this.progressStates.defendedDivorce;
+    }
+
+    return logger.error('No valid case state for ProgressBar page');
```

@dtaylor7 in this scenario should we throw an error? I guess that would display the "Something went wrong" page, instead of just logging and displaying nothing

---

yes you would see 500 error page, not because you throw the error but because the the template would reference content that doesn't exist ( hasn't been returned ).

If you have got to this place, something has gone wrong and you probably should see an error and perhaps contact details to talk to court?

---

Currently it's re-directing back to the root URL `/` if there's no case in session - which is the only time this may happen unless we've missed a state, in which case we should throw an error, that's what I reckon anyway.

---

The other way it could happen if someone changes the state in CCD to something outside of what we expect, but not sure how common/easy that is

---

Alright - so I can put contact details as default case (error state) - which contact details?

---

We're always showing the HMCTS telephone and email.

I'll have a message saying: 'Please contact Courts and Tribunals Service Centre for more information' if the state isn't handled currently.

---

Cool, might be worth checking with UX